src/components/RequestInvite.js

```diff
@@ -0,0 +1,200 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { getRequestInviteOperation } from '../operations';
+import { getMemberId } from '../members';
+import { postOperation, fetchMemberByMidIfNeeded, fetchMemberByUidIfNeeded } from '../actions';
+import YoutubeVideo, { getYoutubeUrlVideoId } from './YoutubeVideo';
+import LogIn from './LogIn';
+import { getAuthMemberDoc, getMemberDocByMid } from '../connectors';
+
+export class RequestInvite extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      videoUrl: '',
+      toUid: '',
+      errorMessage: '',
+      fullName: props.fullName,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.fullName !== this.props.fullName) {
+      this.setState({ fullName: nextProps.fullName });
+    }
+    this.fetchIfNeeded(nextProps);
+  }
+
+  fetchIfNeeded(props) {
+    // TODO this logic very similar to Profile.componentWillReceiveProps.
+    // Decomp/https://reactjs.org/docs/higher-order-components.html/https://github.com/acdlite/recompose
+    if (props.memberId) {
+      this.props.fetchMemberByMidIfNeeded(props.memberId);
+    }
+    if (props.authFirebaseUser) {
+      this.props.fetchMemberByUidIfNeeded(props.authFirebaseUser.uid);
+    }
+  }
+
+  setVideoURL = async (event: any): void => {
+    this.setState({ videoUrl: event.target.value })
+  }
+
+  setFullName = async (event: any): void => {
+    this.setState({ fullName: event.target.value });
+  }
+
+  handleOnSubmit = async (event: any): void => {
+    event.stopPropagation();
+    if (!getYoutubeUrlVideoId(this.state.videoUrl)) {
```

Comment:
the best experience, coming from the user perspective, is to handle validation during `onblur` event of the input.

can make a pr for that once we merge this